scipy/stats/tests/test_continuous_basic.py

```diff
@@ -808,23 +810,54 @@ def test_broadcasting_in_moments_gh12192_regression():
     vals0 = stats.norm.moment(n=1, loc=np.array([1, 2, 3]), scale=[[1]])
     expected0 = np.array([[1., 2., 3.]])
     npt.assert_equal(vals0, expected0)
-    npt.assert_(vals0.shape == expected0.shape)
+    assert vals0.shape == expected0.shape
 
     vals1 = stats.norm.moment(n=1, loc=np.array([[1], [2], [3]]),
                               scale=[1, 2, 3])
     expected1 = np.array([[1., 1., 1.], [2., 2., 2.], [3., 3., 3.]])
     npt.assert_equal(vals1, expected1)
-    npt.assert_(vals1.shape == expected1.shape)
+    assert vals1.shape == expected1.shape
 
     vals2 = stats.chi.moment(n=1, df=[1., 2., 3.], loc=0., scale=1.)
     expected2 = np.array([0.79788456, 1.25331414, 1.59576912])
     npt.assert_allclose(vals2, expected2, rtol=1e-8)
-    npt.assert_(vals2.shape == expected2.shape)
+    assert vals2.shape == expected2.shape
 
     vals3 = stats.chi.moment(n=1, df=[[1.], [2.], [3.]], loc=[0., 1., 2.],
                              scale=[-1., 0., 3.])
     expected3 = np.array([[np.nan, np.nan, 4.39365368],
                           [np.nan, np.nan, 5.75994241],
                           [np.nan, np.nan, 6.78730736]])
     npt.assert_allclose(vals3, expected3, rtol=1e-8)
+    assert vals3.shape == expected3.shape
     npt.assert_(vals3.shape == expected3.shape)
+
+
+# A strategy to generate the shape of a one-dimensional array of length [0..7]
+# Subsequent draws will use the same shape to ensure compatibility.  For fancier
+# tricks see `npst.broadcastable_shapes()` or `mutually_broadcastable_shapes()`.
+shared_shapes = st.shared(st.tuples(st.integers(0, 7)))
```

Comment:
Ideally, to show the advantage of Hypothesis, I think we want this test to go beyond what the tests above can do - show that this works for any distribution and any broadcastable `loc`, `scale`, and shape parameters. (I think it would also be good to nudge hypothesis toward occasionally testing with invalid parameters. I'm not sure how Hypothesis chooses values under the hood, but if we give it such a wide range for scale, is it likely to include in invalid value < 0 in one of the tests?). @tirthasheshpatel @Zac-HD if you are willing to go a little deeper into this, terrific; if not, I understand and I can get to this at some point.

Reply:
I'd recommend having separate tests for always-valid and might-be-invalid inputs; otherwise you eventually and without realising write a test which never checks the valid case :sweat_smile:

For a deeper test, you're going to want [the `mutually_broadcastable_shapes()` strategy](https://hypothesis.readthedocs.io/en/latest/numpy.html#hypothesis.extra.numpy.mutually_broadcastable_shapes) - there's [an example in Numpy for `np.clip()` here](https://github.com/Zac-HD/numpy/blob/4b62c91878386b0a1c837aeb522ce3e2f9d62201/numpy/core/tests/test_numeric.py#L2337-L2384).  Where signatures are consistent you could use `@pytest.mark.parametrize` to supply the moment function - using that to supply some arguments and `@hypothesis.given()` for others works fine - but where signatures differ it's probably easier to write separate tests.

(This is getting pretty far from the original intent to fix a bug though, and I've seen projects take quite a while to settle on using Hypothesis.  Might be worth merging this PR, and then tackling the test changes in a new one?)

Reply:
Thanks for the feedback @Zac-HD and @mdhaber! I agree we should use Hypothesis to its fullest and add a deeper test. I have some time this and next week so I can go a little deeper and add a more comprehensive test.

> This is getting pretty far from the original intent to fix a bug though, and I've seen projects take quite a while to settle on using Hypothesis. Might be worth merging this PR, and then tackling the test changes in a new one?

I also think this would be better. Though I will try to push something or at least confirm locally using Hypothesis that the bug-fix works. I can then propose a PR and we can discuss Hypothesis in detail there. What do you think, @mdhaber?

Reply:
Yes, local test is fine. This PR has been tricky to get right, so a more comprehensive test would give additional confidence that it's ready to be merged. 
And yes, it is actually preferable for the test to go in a separate PR.